pkg/shares/share_builder.go

```diff
@@ -22,28 +22,27 @@ func NewEmptyBuilder() *Builder {
 	}
 }
 
-// Init() needs to be called right after this method
-func NewBuilder(ns appns.Namespace, shareVersion uint8, isFirstShare bool) *Builder {
-	return &Builder{
+// NewBuilder returns a new share builder.
+func NewBuilder(ns appns.Namespace, shareVersion uint8, isFirstShare bool) (*Builder, error) {
+	b := Builder{
 		namespace:      ns,
 		shareVersion:   shareVersion,
 		isFirstShare:   isFirstShare,
 		isCompactShare: isCompactShare(ns),
 	}
+	err := b.init()
```

Comment:
looks good, could even be more concise by having the init in the if phrase like this:

```
if err := b.init(); err != nil {
    return nil, err
}
```

Reply:
109bff6aeee94f5a7bf7ab1b120343ea370ae945